src/Controls/src/Core/Platform/Windows/CollectionView/ItemContentControl.cs

```diff
@@ -289,7 +289,17 @@ protected override WSize MeasureOverride(WSize availableSize)
 
 			var width = ItemWidth == default ? availableSize.Width : ItemWidth;
 			var height = ItemHeight == default ? availableSize.Height : ItemHeight;
-			var measuredSize = base.MeasureOverride(new WSize(width, height));
+			var size = new WSize(width, height);
```

Comment:
There are a couple things about this override that seem a little bit off to me.

- The layout call on line 311 is using VisualElement and it should probably be using `IView`
- It seems wrong that there's a layout call in this code at all, the layout call should probably be inside the `override arrange` ?


Like if I use the following code, the initial size is a bit weird but then once you start clicking the buttons it sizes correctly. 

I'm curious if there's a "Delete more code" solution here. 

```C#
protected override WSize MeasureOverride(WSize availableSize)
{
	if (_renderer == null)
	{
		return base.MeasureOverride(availableSize);
	}

	var width = ItemWidth == default ? availableSize.Width : ItemWidth;
	var height = ItemHeight == default ? availableSize.Height : ItemHeight;
	var size = new WSize(width, height);

	//if (this.Content is FrameworkElement content 
	//	&& CanMeasureContent(content) 
	//	&& (content.DesiredSize.Height != content.Height || content.DesiredSize.Width != content.Width))
	//{
	//	// Measure content to update its desired size if its Height/Width just changed
	//	content.Measure(size);
	//}

	var theSize = _renderer.VirtualView.Measure(size.Width, size.Height);
	var measuredSize = base.MeasureOverride(theSize.ToPlatform());
	var finalWidth = Max(measuredSize.Width, ItemWidth);
	var finalHeight = Max(measuredSize.Height, ItemHeight);
	var finalSize = new WSize(finalWidth, finalHeight);
	var frameworkElement = Content as FrameworkElement;
	var visualElement = _renderer.VirtualView as IView;
	var margin = _renderer.VirtualView.Margin;

	frameworkElement.Margin = WinUIHelpers.CreateThickness(margin.Left, margin.Top, margin.Right, margin.Bottom);
	//visualElement.Arrange(new Rect(0, 0, finalWidth, finalHeight));

	/*if (CanMeasureContent(frameworkElement))
	{
		frameworkElement.Measure(finalSize);
	}*/

	return finalSize;
}
```

Reply:
Does this work?

```C#
/// <inheritdoc/>
protected override WSize ArrangeOverride(WSize finalSize)
{
	if (_renderer == null)
	{
		return base.ArrangeOverride(finalSize);
	}

	/*var width = ItemWidth == default ? finalSize.Width : ItemWidth;
	var height = ItemHeight == default ? finalSize.Height : ItemHeight;
	var size = new WSize(width, height);*/

	var result =  base.ArrangeOverride(finalSize);
	return result;
}

/// <inheritdoc/>
protected override WSize MeasureOverride(WSize availableSize)
{
	if (_renderer == null)
	{
		return base.MeasureOverride(availableSize);
	}

	var width = ItemWidth == default ? availableSize.Width : ItemWidth;
	var height = ItemHeight == default ? availableSize.Height : ItemHeight;
	var size = new WSize(width, height);
	var frameworkElement = Content as FrameworkElement;


	// Do we really need this margin code?
	var margin = _renderer.VirtualView.Margin;
	frameworkElement.Margin = WinUIHelpers.CreateThickness(margin.Left, margin.Top, margin.Right, margin.Bottom);

	var measuredSize = base.MeasureOverride(size);
	var finalWidth = Max(measuredSize.Width, ItemWidth);
	var finalHeight = Max(measuredSize.Height, ItemHeight);
	var finalSize = new WSize(finalWidth, finalHeight);

	return finalSize;
}
```

Reply:
Alternate thoughts
https://github.com/dotnet/maui/pull/18932